app/server/lib/uploads.ts

```diff
@@ -426,9 +426,28 @@ export async function fetchDoc(
   // The backend needs to be well configured for this to work.
   const { selfPrefix, docWorker } = await getDocWorkerInfoOrSelfPrefix(docId, docWorkerMap, server.getTag());
   const docWorkerUrl = docWorker ? docWorker.internalUrl : getUrlFromPrefix(server.getHomeInternalUrl(), selfPrefix);
+  const apiBaseUrl = docWorkerUrl.replace(/\/*$/, '/');
+
+  // Documents with external attachments can't be copied right now. Check status and alert the user.
+  const transferStatusResponse = await fetch(
+      new URL(`/api/docs/${docId}/attachments/transferStatus`, apiBaseUrl).href,
+    {
+      headers: {
+        ...headers,
+        'Content-Type': 'application/json',
+      }
+    }
+  );
+  if (!transferStatusResponse.ok) {
+    throw new ApiError(await transferStatusResponse.text(), transferStatusResponse.status);
+  }
+  const attachmentsLocation: DocAttachmentsLocation = (await transferStatusResponse.json()).locationSummary;
+  if (attachmentsLocation !== 'internal' && attachmentsLocation !== 'none') {
+    throw new ApiError("Cannot copy a document with external attachments", 400);
```

Comment:
In the options object you have `template` parameter. When it is passed here as `true` it means that user wants to copy the structure, not the data. So in this case it should be allowed.

Reply:
This is a really good catch, thanks :)

Reply:
This should be sorted now :)